sycl/source/device.cpp

```diff
@@ -134,15 +133,32 @@ bool device::has_extension(const std::string &extension_name) const {
   return impl->has_extension(extension_name);
 }
 
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+template <typename Param>
+detail::ABINeutralT_t<typename detail::is_device_info_desc<Param>::return_type>
+device::get_info_impl() const {
+  return detail::convert_to_abi_neutral(impl->template get_info<Param>());
+}
+#else
 template <typename Param>
-typename detail::is_device_info_desc<Param>::return_type
+detail::ABINeutralT_t<typename detail::is_device_info_desc<Param>::return_type>
 device::get_info() const {
-  return impl->template get_info<Param>();
+  static_assert(
+      std::is_same_v<detail::ABINeutralT_t<typename detail::is_device_info_desc<
+                         Param>::return_type>,
+                     typename detail::is_device_info_desc<Param>::return_type>);
+  return detail::convert_to_abi_neutral(impl->template get_info<Param>());
```

Comment:
Drop the `#if` and its `#else` branch...

Reply:
Thanks. Done.

Reply:
Actually, this caused lots of ABI symbols to be changed.
And currently, sycl_symbols_linux.dump contains lots of changes.
For example, we changed from
```
_ZNK4sycl3_V18platform8get_infoINS0_4info8platform10extensionsEEENS0_6detail21is_platform_info_descIT_E11return_typeEv
```
to 
```
_ZNK4sycl3_V18platform13get_info_implINS0_4info8platform10extensionsEEENS0_6detail21is_platform_info_descIT_E11return_typeEv
```
Is this considered an ABI breaking change or not?

Reply:
You can do
```
template <...>
<return type>
#if 
device::get_info_impl
#else
device::get_info
#endif
const { ... }
```

Reply:
Thanks a lot. I fixed it as you suggested.